provisioning/service/tests/TestAssert.cs

```diff
@@ -38,6 +38,11 @@ public static class TestAssert
             throw new AssertFailedException("{0}. Expected {1} exception but no exception is thrown".FormatInvariant(errorMessage, typeof(TException).ToString()));
         }
 
+        internal static void Throws<T>(Func<Task> p)
+        {
+            throw new NotImplementedException();
+        }
+
         public static TException Throws<TException>(Func<Task> action, string errorMessage = null) where TException : Exception
         {
             return Throws<TException>(() => action().Wait(), errorMessage);
```

Comment:
Waiting on Tasks is likely to cause deadlocks.

Reply:
removed